server-common/src/main/java/org/apache/cassandra/sidecar/common/server/cluster/locator/TokenRange.java

```diff
@@ -75,18 +75,83 @@ public TokenRange(long start, long end)
 
     public TokenRange(BigInteger start, BigInteger end)
     {
-        this.start = start;
-        this.end = end;
+        this(Token.from(start), Token.from(end));
+    }
+
+    public TokenRange(Token start, Token end)
+    {
+        this.range = Range.openClosed(start, end);
+    }
+
+    /**
+     * @return start token. It is not enclosed in the range.
+     */
+    public Token start()
+    {
+        return range.lowerEndpoint();
+    }
+
+    /**
+     * @return end token. It is the last token enclosed in the range.
+     */
+    public Token end()
+    {
+        return range.upperEndpoint();
+    }
+
+    /**
+     * @return the first token enclosed in the range
+     */
+    @Nullable
+    public Token firstToken()
+    {
+        if (range.isEmpty())
+        {
+            return null;
+        }
+        return range.lowerEndpoint().increment();
```

Comment:
Curious. Why returning the increment here? Shouldn't it return `range.lowerEndpoint()`?

Reply:
The token range is always open-closed. The lower end is not included. So the first token is the lower end + 1.